### PR TITLE
Harmony 1278 output file size

### PR DESCRIPTION
**Pull request by indiejames**
## Jira Issue ID
HARMONY-1278

## Description
Adds  dB migrations and code to lookup/store granule sizes to support batching.

## Local Test Steps
* Deploy to the sandbox
* Issue a harmony query to deployed environment
* Connect to dB and select work items
* Verify that the work items have a value for the outputGranuleSizes column

## PR Acceptance Checklist
* [x] Acceptance criteria met
* [x] Tests added/updated (if needed) and passing
* [ ] ~Documentation updated (if needed)~

### CODE REVIEW

**Review by chris-durbin — APPROVED** [review]
Tested in sandbox

File: tasks/query-cmr/app/query.ts [hunk 1]
```diff
@@ -42,8 +43,16 @@ async function querySearchAfter(
...
   let totalGranulesSize = 0;
+  const outputGranuleSizes = [];
   const catalogs = cmrResponse.granules.map((granule) => {
     const granuleSize = granule.granule_size ? parseFloat(granule.granule_size) : 0;
+    let granuleSizeInBytes = granuleSize * 1024 * 1024;
+    // NaN will fail the first check
+    if (granuleSizeInBytes != granuleSizeInBytes || granuleSizeInBytes < 0) {
+      granuleSizeInBytes = 0;
+    }
+    logger.info(`Granule size: ${granuleSizeInBytes}`);
```
[the remark this answers is not in the document]
Comment (indiejames) [reply tasks/query-cmr/app/query.ts:54]: Done.

File: test/util/crypto.ts [hunk 2]
```diff
@@ -18,8 +18,9 @@ describe('util/crypto', function () {
       it('decrypts ciphertext into the correct plaintext', function () {
         const sharedSecretKey = '_THIS_IS_MY_32_CHARS_SECRET_KEY_';
         const encrypter = createEncrypter(sharedSecretKey);
-        const plaintext = 'The secret code is the number 7';
```
[the remark this answers is not in the document]
Comment (indiejames) [reply test/util/crypto.ts:21]: Done.

File: app/backends/workflow-orchestration.ts [hunk 3]
```diff
@@ -477,19 +480,107 @@ async function updateWorkItemCounts(
...
+  logger.debug(`ContentLength: ${result}`);
+  return result;
+}
+
+/**
+ * Read and parse a STAC catalog and return the links to the data items
+ * 
+ * @param s3Url - the s3 url of the catalog
+ */
+export async function readSTACCatalog(s3Url: string, logger: Logger): Promise<string[]> {
```
[the remark this answers is not in the document]
Comment (indiejames) [reply app/backends/workflow-orchestration.ts]: I renamed it to readCatalogLinks and am  leaving it here as it is used nowhere else. The only reason I export it is so we can stub it in tests.

**Review by vinnyinverso — APPROVED** [review]

**Pull request merged**